### PR TITLE
FEATURE: Back button for schema theme settings

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Continue from https://github.com/discourse/discourse/pull/25673.

This PR adds a back button to the navigation tree of the schema.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/components/admin-theme-setting-schema.gjs [hunk 1]
```diff
@@ -25,17 +27,16 @@ class Tree {
 @tagName("")
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/components/admin-theme-setting-schema.gjs:27]: Could you remove this while here? 🙂 (it's a no-op, that only applies to non-glimmer components)

File: app/assets/javascripts/admin/addon/components/admin-theme-setting-schema.gjs [hunk 2]
```diff
@@ -25,17 +27,16 @@ class Tree {
...
 export default class AdminThemeSettingSchema extends Component {
   @tracked activeIndex = 0;
+  @tracked backButtonText = null;
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/components/admin-theme-setting-schema.gjs:30]: There's no need to initialize null fields

```suggestion
  @tracked backButtonText;
```

File: app/assets/javascripts/discourse/tests/integration/components/admin-theme-setting-schema-test.gjs [hunk 3]
```diff
@@ -133,48 +140,48 @@ module(
         <AdminThemeSettingSchema @schema={{schema}} @data={{data}} />
       </template>);
 
-      const tree = queryRenderedTree();
+      const tree = new TreeFromDOM();
 
-      assert.equal(tree.length, 2);
-      assert.true(tree[0].active, "the first node is active");
-      assert.false(tree[1].active, "other nodes are not active");
+      assert.equal(tree.nodes.length, 2);
```
Comment (CvX) [inline app/assets/javascripts/discourse/tests/integration/components/admin-theme-setting-schema-test.gjs:145]: We don't lint against it yet, but `assert.strictEqual()` is preferred over `assert.equal()`

**Review by tgxworld — APPROVED** [review]
The PR looks good to me and is simple to follow 👍 Great work! 

There are some minor comments which I think should be addressed before merging though.

**Pull request merged**